charts/vector/templates/statefulset.yaml

```diff
@@ -6,7 +6,7 @@ metadata:
   labels:
     {{- include "vector.labels" . | nindent 4 }}
 spec:
-  {{- if not .Values.autoscaling.enabled }}
+  {{- if and (not .Values.autoscaling.enabled) (not .Values.keda.enabled) }}
```

Comment:
Same as above: I think this should be an `or`? If either of them is enabled we don't want to include the replicas key.